app/models/work_packages/scopes/include_spent_time.rb

```diff
@@ -31,12 +31,11 @@ module WorkPackages::Scopes::IncludeSpentTime
 
   class_methods do
     def include_spent_time(user, work_package = nil)
-      query = join_time_entries(user)
-
-      scope = left_join_self_and_descendants(user, work_package)
-              .joins(query.join_sources)
+      scope = with(visible_time_entries_cte.name => allowed_to_view_time_entries(user))
+              .left_join_self_and_descendants(user, work_package)
```

Comment:
This should remove the double CTE inclusion:
```suggestion
      scope = left_join_self_and_descendants(user, work_package)
              .with(visible_time_entries_cte.name => allowed_to_view_time_entries(user))
```

Reply:
Thanks @toy , that helps indeed.